include/seqan3/range/view/take_until.hpp

```diff
@@ -0,0 +1,453 @@
+    using const_iterator    = std::conditional_t<const_iterable, iterator, void>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    view_take_until() = default;
+    constexpr view_take_until(view_take_until const & rhs) = default;
+    constexpr view_take_until(view_take_until && rhs) = default;
+    constexpr view_take_until & operator=(view_take_until const & rhs) = default;
+    constexpr view_take_until & operator=(view_take_until && rhs) = default;
+    ~view_take_until() = default;
+
+    /*!\brief Construct from another range.
+     * \param[in] _urange The underlying range.
+     * \param[in] _fun    The functor that acts as termination criterium.
+     */
+    view_take_until(urng_t _urange, fun_t _fun)
+        : urange{std::move(_urange)}, fun{std::forward<fun_t>(_fun)}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to seqan3::view_take_until::end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {ranges::begin(urange), static_cast<fun_t &>(fun)};
+    }
+
+    //!\copydoc begin()
+    const_iterator begin() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cbegin(urange), static_cast<fun_t const &>(fun)};
+    }
+
+    //!\copydoc begin()
+    const_iterator cbegin() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cbegin(urange), static_cast<fun_t const &>(fun)};
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    sentinel_type end() noexcept
+    {
+        return {ranges::end(urange)};
+    }
+
+    //!\copydoc end()
+    sentinel_type end() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cend(urange)};
+    }
+
+    //!\copydoc end()
+    sentinel_type cend() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cend(urange)};
+    }
+    //!\}
+
+    /*!\brief Convert this view into a container implicitly.
+     * \tparam container_t Type of the container to convert to; must model seqan3::container_concept and it's
+     *                     seqan3::reference_t must model seqan3::common_reference_concept with `reference`.
+     * \returns This view converted to container_t.
+     */
+    template <container_concept container_t>
+    operator container_t()
```

Comment:
The operator is called `_t` even though it returns an object and not a type.. I'd call it `get_container` or just `conatainer`.

Reply:
It's a user defined conversion operator so obviously it's called like the type it's converting to :wink: